WolvenKit/Views/Editors/AutomaticNodifyEditor.cs

```diff
@@ -0,0 +1,230 @@
+                        graph.Nodes.Add(msaglNode);
+
+                        // create a lookup between the inputs/outputs & the node itself
+                        foreach (var input in nvm.Inputs)
+                        {
+                            socketNodeLookup[input.GetHashCode()] = nvm.GetHashCode();
+                        }
+                        foreach (var output in nvm.Outputs)
+                        {
+                            socketNodeLookup[output.GetHashCode()] = nvm.GetHashCode();
+                        }
+                    }
+                }
+            }
+
+            foreach (INodeConnection connection in Connections)
+            {
+                if (socketNodeLookup.ContainsKey(connection.Source.GetHashCode()) &&
+                    socketNodeLookup.ContainsKey(connection.Destination.GetHashCode()))
+                {
+                    var source = socketNodeLookup[connection.Source.GetHashCode()];
+                    var dest = socketNodeLookup[connection.Destination.GetHashCode()];
+                    var edge = new Edge(nodes[source], nodes[dest]);
+                    edge.TargetPort = new FloatingPort(
+                        edge.Target.BoundaryCurve,
+                        new Microsoft.Msagl.Core.Geometry.Point(-200, 0));
+                    edge.SourcePort = new FloatingPort(
+                        edge.Source.BoundaryCurve,
+                        new Microsoft.Msagl.Core.Geometry.Point(200, 0));
+                    graph.Edges.Add(edge);
+                }
+            }
+
+            var settings = new SugiyamaLayoutSettings
+            {
+                Transformation = PlaneTransformation.Rotation(Math.PI / 2),
+                NodeSeparation = 30
+                //EdgeRoutingSettings = { EdgeRoutingMode = EdgeRoutingMode.Spline }
+            };
+            var layout = new LayeredLayout(graph, settings);
+            //var settings = new MdsLayoutSettings()
+            //{
+
+            //};
+            //var layout = new MdsGraphLayout(settings, graph);
+            try
+            {
+                layout.Run();
+
+                foreach (var item in ItemsHost.Children)
+                {
+                    if (item is ItemContainer ic)
+                    {
+                        if (ic.DataContext is INode nvm && nodes.ContainsKey(nvm.GetHashCode()))
+                        {
+                            var node = nodes[nvm.GetHashCode()];
+                            nvm.Location = new Point(
+                                node.Center.X - graph.BoundingBox.Center.X - ic.ActualWidth / 2,
+                                node.Center.Y - graph.BoundingBox.Center.Y - ic.ActualHeight / 2);
+                        }
+                    }
+                }
+
+                // center to 0,0 since we're subtracting the graph's center from each location
+                BringIntoView(new Point(0, 0));
```

Comment:
I don't see why this would be necessary. You never move the editor's `Location`.

Reply:
I think this was made obsolete by some changes elsewhere - thanks!